WorkloadManagementSystem/PilotAgent/PilotLogger/PilotLogger.py

```diff
@@ -0,0 +1,211 @@
+    FLAGS, check constructor for current set
+    """
+
+    if flag in self.FLAGS:
+      return True
+    return False
+
+  def _isCorrectStatus( self, status ):
+
+    """
+    Method checks if the flag corresponds to one of the predefined
+    STATUSES, check constructor for current set
+    """
+
+    if status in self.STATUSES:
+      return True
+    return False
+
+  def _connect(self):
+    """ Methods connects to RabbitMQ and returns connection
+    handler or None in case of connection down
+    """
+    try:
+      connection = pika.BlockingConnection(
+      pika.ConnectionParameters( self.networkCfg['host'] )
+        )
+    except pika.exceptions.AMQPConnectionError, argument:
+      print 'Connection error: %r' %(argument,)
+      return None
+    else:
+      return connection
+
+
+  def _sendAllLocalMessages(self, connect_handler, flag = 'info' ):
+    """
+    Method retrives all messages from the local storage
+    and sends it.
+    """
+
+    channel = connect_handler.channel()
+    channel.exchange_declare(
+                exchange = self.networkCfg['exchangeName'],
+                type = self.networkCfg['exchangeType']
+                )
+    queue =readMessagesFromFileAndEraseFileContent()
+    while not queue.empty():
+      msg = queue.get()
+      channel.basic_publish(
+                          exchange = self.networkCfg['exchangeName'],
+                          routing_key = flag,
+                          body = msg,
+                          )
+      print " [x] Sent %r %r" % ( type, msg )
+
+
+  def _sendMessage( self, msg, flag ):
+    """Method first copies the message content to the
+       local storage, then itchecks if the connection
+       to RabbitMQ server is up,
+       If it is the case it sends all messages stored
+       locally.  The string flag can be used as routing_key,
+       it can contain:  'info', 'warning', 'error',
+       'debug'. If the connection is down, the method
+       does nothing and returns False
+       @return: False in case of any errors, True otherwise
+    """
+
+    if not self._isCorrectFlag( flag ):
+      return False
+    saveMessageToFile(msg)
+    connection = self._connect()
+    if not connection:
+      return False
+    self._sendAllLocalMessages(connection, flag)
+    connection.close()
+    return True
+
+  def sendMessage( self, minorStatus, flag = 'info', status='Installing' ):
+    """
+    Method sends the message after
+    creating the correct format:
+    including content, timestamp, status, minor status and the uuid
+    of the pilot
+    @return : False in case of any errors, True otherwise
+    """
+    print "im sending"
+    if not self._isCorrectFlag( flag ):
+      return False
+    print "really?"
+    print status
+    if not self._isCorrectStatus( status ):
+      return False
+    print "really?"
+    myUUID = getPilotUUIDFromFile()
+    message = generateDict(
+        myUUID,
+        '',  # pilotID is unknown for a moment
```

Comment:
And it will always remain unknown: the only known thing is the UUID (which can be the so-called "Pilot Reference", AKA $VM_UUID for VMs or other identifiers.